### ISSUE TITLE
GDIP model as a regularizer

### ISSUE THREAD

**Issue by Unclechan97** [Gatedip/GDIP-Yolo#8]
First of all, thank you very much for providing the paper idea. I have a few questions to ask. The code for using the GDIP model as a regularizer seems to be different from the original paper. Additionally, it appears that the code does not include the reconstruction loss.

**Comment by sanketkalwar** [comment]
@Unclechan97, We haven't yet provided our model training code as it is a funded project. And you can use nn.L1Loss() and nn.MSELoss() for the reconstruction loss. You can also read the paper to get details of hyperparameters used to train the model. I hope this helps!

**Comment by Unclechan97** [comment]
I feel that this piece of code is somewhat different from the description in the original paper. Here, the output of GDIP becomes the input of YOLOv3, and GDIP has become a part of YOLOv3. So, how should it be removed during inference?
class Yolov3MGatedDIP(torch.nn.Module):
    def __init__(self):
        super(Yolov3MGatedDIP,self).__init__()
        self.mgdip = MultiLevelGDIP(256,7)
        self.yolov3 = Yolov3()
    
    def forward(self,x):
        out_x,_,gates_list = self.mgdip(x)
        p,p_d = self.yolov3(out_x)
        return out_x,gates_list,p,p_d
